crates/dladm/src/lib.rs

```diff
@@ -140,7 +142,11 @@ impl Handle {
             state.n_seen += 1;
 
             if (*macaddr).ma_addrlen == (ETHERADDRL as u32) {
-                state.mac.copy_from_slice(&(*macaddr).ma_addr[..ETHERADDRL]);
+                let ma_addr = slice::from_raw_parts(
+                    ptr::addr_of!((*macaddr).ma_addr).cast::<u8>(),
```

Comment:
i think we can (should?) change this so that `Arg` has a `&'a mut [u8; ETHADDRL]` so we can side-step this and the question of "will copy_from_slice blow up somehow", but that's neither important nor better here :woman_shrugging: 

fwiw this apparently can also be `&raw const (*macaddr).ma_addr)` which i did not realize is a modern replacement for `ptr::addr_of!()`

Reply:
apparently 1.82! neat